src/resource.rs

```diff
@@ -206,18 +215,35 @@ impl HalResource {
     pub fn get_links(&self, name: &str) -> Option<&Vec<Box<HalLink>>> {
         match self.links.get(name) {
             Some(link) => Some(link.many()),
-            None => None
+            None => None,
         }
     }
 
-    pub fn with_resource(&mut self, name: &str, resource: &HalResource) -> &mut Self
-    {
+    pub fn with_resource(&mut self, name: &str, resource: &HalResource) -> &mut Self {
         match self.embedded.entry(name.to_string()) {
             Entry::Vacant(entry) => {
                 let mut resources = OneOrMany::new();
                 resources.push(&resource.clone());
                 entry.insert(resources);
-            },
+            }
+            Entry::Occupied(mut entry) => {
+                let mut content = entry.get_mut(); //&mut HalLinks
+                content.push(&resource.clone());
+            }
+        }
+        self
+    }
+
+    pub fn with_resources(&mut self, name: &str, resource: &HalResource) -> &mut Self {
+        match self.embedded.entry(name.to_string()) {
+            Entry::Vacant(entry) => {
+                let mut resources = OneOrMany {
+                    content: Vec::new(),
+                    force_many: true,
+                };
```

Comment:
I think it would be preferable to call the constructor (new) and then call force_many() on the result, or create a new constructor that sets force_many to true, rather than building the struct directly.